### PR TITLE
Updated specs and elements from the schema

### PR DESCRIPTION
**Pull request by jc-berger**
For issue [#158: Update Stack Author's Guide](https://github.com/devfile/api/issues/158). 

@elsony, tagging you for your awareness since we've worked together on this issue. Please let me know who's best to do a technical review, and I'll be glad to work with that developer. 

Please note, I'll send up a new commit when I am done editing each separate doc. I'll squash the commits to make peer review easier. 

Thanks!

### CODE REVIEW

**Comment by rkratky** [comment]
@jc-berger, please, remove all the `.DS_Store` files from the PR (they are macOS-specific folder-attribute files that have no place in the repo).

To avoid having them committed together with your changes next time, add the `.DS_Store` file to your `.gitignore` file. See, for example, https://timleland.com/global-git-ignore/, for instructions on how to configure a user-wide `.gitignore` file on a Mac.

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 1]
```diff
@@ -758,9 +759,10 @@ projects:
   - name: my-go-project
     clonePath: go/src/github.com/acme/my-go-project
     git:
-      location: https://github.com/acme/my-go-project.git
+      remotes: https://github.com/acme/my-go-project.git
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:762]: Should be (make similar changes to other remotes references to add `origin` like this example):
```
      remotes: 
        origin: "https://github.com/acme/my-go-project.git"
```

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 2]
```diff
@@ -842,9 +844,10 @@ projects:
...
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:847]: Need to add origin (see above)

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 3]
```diff
@@ -892,10 +895,10 @@ projects:
...
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:898]: Need to add origin (see above)

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 4]
```diff
@@ -934,9 +937,10 @@ projects:
...
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:940]: Need to add origin (see above)

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 5]
```diff
@@ -968,25 +972,25 @@ projects:
   - name: my-maven-project
     clonePath: maven/src/github.com/acme/my-maven-project
     git:
-      location: https://github.com/acme/my-maven-project.git
+      remotes: https://github.com/acme/my-maven-project.git
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:975]: Need to add origin (see above)

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 6]
```diff
@@ -1004,29 +1008,29 @@ projects:
...
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:1011]: Need to add origin (see above)

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 7]
```diff
@@ -1106,7 +1110,7 @@ metadata:
 projects:
   - name: petclinic
     git:
-      location: 'https://github.com/che-samples/web-java-spring-petclinic.git'
+      remotes: 'https://github.com/che-samples/web-java-spring-petclinic.git'
```
Comment (elsony) [inline docs/modules/ROOT/partials/ref_devfile-reference.adoc:1113]: Need to add origin (see above)

**Review by elsony — CHANGES_REQUESTED** [review]

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 8]
```diff
@@ -83,7 +83,7 @@ projects:
   - name: my-project
```
[the remark this answers is not in the document]
Comment (jc-berger) [reply docs/modules/ROOT/partials/ref_devfile-reference.adoc:83]: Addressed, so resolved convo.

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 9]
```diff
@@ -31,7 +31,7 @@ Project source consists of one of the following types: `git`, `github` and `zip`
 ----
 source:
```
[the remark this answers is not in the document]
Comment (jc-berger) [reply docs/modules/ROOT/partials/ref_devfile-reference.adoc:32]: Addressed, so resolved convo.

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc [hunk 10]
```diff
@@ -615,10 +616,10 @@ projects:
   - name: my-go-project
     clonePath: go/src/github.com/acme/my-go-project
     git:
-      location: https://github.com/acme/my-go-project.git
+      remotes: https://github.com/acme/my-go-project.git
```
[the remark this answers is not in the document]
Comment (jc-berger) [reply docs/modules/ROOT/partials/ref_devfile-reference.adoc:619]: Addressed, so resolved convo.

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc (hunk 1 above)
Comment (jc-berger) [reply docs/modules/ROOT/partials/ref_devfile-reference.adoc:762]: Addressed, so resolved convo.

File: docs/modules/ROOT/partials/ref_devfile-reference.adoc (hunk 2 above)
Comment (jc-berger) [reply docs/modules/ROOT/partials/ref_devfile-reference.adoc:847]: Addressed, so resolved convo.

**Comment by jc-berger** [comment]
@elsony, thanks for the review :) 

I made a new commit `fixed remotes origin`. Let me know what you think, thanks!

**Review by elsony — APPROVED** [review]
/lgtm

**Pull request merged**